x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -234,6 +235,34 @@ static DeprecationIssue checkTemplatesWithMultipleTypes(ClusterState state) {
         );
     }
 
+    static DeprecationIssue checkTemplatesWithCustomTypes(ClusterState state) {
+        Set<String> templatesWithCustomTypes = new TreeSet<>();
+        state.getMetadata().getTemplates().forEach((templateCursor) -> {
+            String templateName = templateCursor.key;
+            ImmutableOpenMap<String, CompressedXContent> mappings = templateCursor.value.mappings();
+            if (mappings != null && mappings.size() == 1) {
+                String typeName = mappings.iterator().next().key;
+                if ("_doc".equals(typeName) == false) {
```

Comment:
Tiny comment, this could be `MapperService.SINGLE_MAPPING_NAME`.

Reply:
Oops, I'd meant to go back and find that constant!